examples/dataflow-csvio/src/main/java/com/google/example/csvio/SortContextualHeadersAndRows.java

```diff
@@ -0,0 +1,427 @@
+    }
+
+    public PCollection<Row> getHeaders() {
+      return headers;
+    }
+
+    public PCollection<Row> getRows() {
+      return rows;
+    }
+
+    @Override
+    public Pipeline getPipeline() {
+      return this.pipeline;
+    }
+
+    @Override
+    public Map<TupleTag<?>, PValue> expand() {
+      return new HashMap<>() {
+        {
+          this.put(HEADERS, headers);
+          this.put(ROWS, rows);
+        }
+      };
+    }
+
+    public CSVIO.Read.Result apply(PTransform<? super Result, CSVIO.Read.Result> t) {
+      return t.expand(this);
+    }
+
+    @Override
+    public void finishSpecifyingOutput(
+        String transformName, PInput input, PTransform<?, ?> transform) {}
+  }
+
+  /**
+   * A {@link PTransform} responsible for bundling the raw {@link ContextualTextIO#read()} {@link
+   * Row} {@link PCollection} output into the corresponding CSV headers and rows in the setting
+   * where a fixed header position is expected.
+   */
+  static class SortFixedHeaderAndRows extends PTransform<PCollection<Row>, Result> {
+
+    private static final String TAG_BASE =
+        SortContextualHeadersAndRows.TAG_BASE + "/" + SortFixedHeaderAndRows.class.getSimpleName();
+
+    private final CSVIOReadConfiguration configuration;
+
+    SortFixedHeaderAndRows(CSVIOReadConfiguration configuration) {
+      this.configuration = configuration;
+    }
+
+    @Override
+    public Result expand(PCollection<Row> input) {
+      PCollectionView<Long> headerPositionView =
+          input
+              .getPipeline()
+              .apply(TAG_BASE + "/HeaderPositionView", Create.of(configuration.getHeaderPosition()))
+              .apply(View.asSingleton());
+      PCollectionTuple pct =
+          input.apply(
+              TAG_BASE + "/" + SortMappedFixedPositionHeadersAndRowsFn.class.getSimpleName(),
+              ParDo.of(new SortFixedPositionHeadersAndRowsFn(headerPositionView))
+                  .withSideInput("", headerPositionView)
+                  .withOutputTags(HEADERS, TupleTagList.of(ROWS)));
+
+      return new Result(pct);
+    }
+  }
+
+  /**
+   * A {@link PTransform} responsible for bundling the raw {@link ContextualTextIO#read()} {@link
+   * Row} {@link PCollection} output into the corresponding CSV headers and rows in the setting
+   * where a regular expression header matching is expected.
+   */
+  static class SortMatchedHeaderAndRows extends PTransform<PCollection<Row>, Result> {
+
+    private static final String TAG_BASE =
+        SortContextualHeadersAndRows.TAG_BASE
+            + "/"
+            + SortMatchedHeaderAndRows.class.getSimpleName();
+    private final CSVIOReadConfiguration configuration;
+
+    SortMatchedHeaderAndRows(CSVIOReadConfiguration configuration) {
+      this.configuration = configuration;
+    }
+
+    @Override
+    public Result expand(PCollection<Row> input) {
+      PCollectionView<String> matchHeaderRegexView =
+          input
+              .getPipeline()
```

Comment:
Same comments as earlier on usage of getPipeline as well as passing getHeaderMatchRegex directly instead of building PCollectionView